### PR TITLE
Updates for iOS in-context signup support

### PR DESCRIPTION
**Pull request by alistairjcbrown**
**Reviewer:** @GioSensation 
**Asana:** https://app.asana.com/0/0/1204874198702028/f

## Description
Passes a new flag when autofill is opened to let the native side know if in-context signup is available.
On successful response, updates autofill state and re-renders icon.

## Steps to test

Test in conjunction with https://github.com/duckduckgo/iOS/compare/anya/ios-incontext-email-promotion

### CODE REVIEW

File: src/Form/Form.js [hunk 1]
```diff
@@ -510,7 +510,8 @@ class Form {
             if (!canBeInteractedWith(input)) return
 
             // Checks for pointerdown event. Needed for positioning when fields are within an iframe
-            if (e.type === 'pointerdown') {
+            // TODO: Confirm this is an ok change -- by not calling `getMainClickCoords` we're no longer checking if the event is trusted
```
Comment (alistairjcbrown) [inline src/Form/Form.js:513]: Added this `todo` here as the comment didn't match the implementation. I've updated it to check for fields within an iframe, but I want to make sure we're not losing other checks by doing this.

File: src/DeviceInterface/AppleDeviceInterface.js [hunk 2]
```diff
@@ -140,15 +140,7 @@ class AppleDeviceInterface extends InterfacePrototype {
...
-                // Remove decorations before refreshing data to make sure we
-                // remove the currently set icons
-                this.activeForm?.removeAllDecorations()
-
-                // Update for any state that may have changed
-                await this.refreshData()
-
-                // Add correct icons and behaviour
-                this.activeForm?.recategorizeAllInputs()
+                await this.updateForStateChange()
```
Comment (alistairjcbrown) [inline src/DeviceInterface/AppleDeviceInterface.js:143]: Extracted this out to a named function on `src/DeviceInterface/InterfacePrototype.js` so we can reuse it

File: src/DeviceInterface/AppleDeviceInterface.js [hunk 3]
```diff
@@ -274,7 +266,8 @@ class AppleDeviceInterface extends InterfacePrototype {
     async getAlias () {
         const {alias} = await this.deviceApi.request(new GetAlias({
             requiresUserPermission: !this.globalConfig.isApp,
-            shouldConsumeAliasIfProvided: !this.globalConfig.isApp
+            shouldConsumeAliasIfProvided: !this.globalConfig.isApp,
+            isIncontextSignupAvailable: this.inContextSignup.isAvailable()
```
Comment (alistairjcbrown) [inline src/DeviceInterface/AppleDeviceInterface.js:270]: Sharing in-context signup availability with native side

File: src/InContextSignup.js [hunk 4]
```diff
@@ -54,13 +54,18 @@ export class InContextSignup {
         const activeInput = this.device.activeForm?.activeInput
         activeInput?.blur()
 
-        // Focus on the active input to open the tooltip
-        const focusActiveInput = () => {
+        // Select the active input to open the tooltip
+        const selectActiveInpout = () => {
```
Comment (GioSensation) [inline src/InContextSignup.js:58]: Not sure if it's ready for review, but I've noticed the typo in `selectActiveInpout`.

**Review by GioSensation — APPROVED** [review]
The code looks good and the flow is awesome 💪! I've just posted a somewhat edgey case on Asana. Not sure if we want to tackle it. If it's simple enough, I think we should. Other than that, just a typo reported in my previous comment.

**Comment by CDRussell** [comment]
What's happening with this? Anything left to do before it's merged?

**Pull request merged**